airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/ctid/CtidCountStateManager.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.postgres.ctid;
+
+import static io.airbyte.integrations.source.postgres.PostgresUtils.isCdc;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.cdk.integrations.source.relationaldb.state.SourceStateMessageProducer;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.v0.AirbyteStreamState;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A state manager for non resumable full refresh. It does not produce any meaningful state, but
+ * just serves as feeding into SourceStateIterator so we can have counts.
+ */
+public class CtidCountStateManager implements SourceStateMessageProducer<AirbyteMessage> {
```

Comment:
Can we have "non-resumable" in the name somewhere?

Reply:
Sure. While doing this change I actually realized I can refactor this into CDK to make things much cleaner!